### PR TITLE
SCRUM-4190

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/GeneOntologyAnnotationService.java [hunk 1]
```diff
@@ -106,14 +109,24 @@ private Long getGOID(GeneOntologyAnnotationDTO uiEntity) {
...
 
-	private Species getSpecies(String orgAbbreviation) {
-		if (species != null) {
+	private List<Species> getSpecies(String orgAbbreviation) {
+		if (CollectionUtils.isNotEmpty(species)) {
 			return species;
 		}
+		if (orgAbbreviation.equalsIgnoreCase("XB")) {
+			species = new ArrayList<>();
+			species.add(getSingleSpecies("XBXL"));
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/services/GeneOntologyAnnotationService.java:118]: The whole getSingleSpecies method could be replaced here with `speciesDAO.findByField("displayName", "XB").getSingleResult();`

File: src/main/resources/db/migration/v0.38.0.3__gaf-load.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+update bulkurlload set bulkloadurl = 'https://download.alliancegenome.org/2.3.0/GAF/HUMAN/GAF_HUMAN_1.gaf'
```
Comment (markquintontulloch) [inline src/main/resources/db/migration/v0.38.0.3__gaf-load.sql]: Need to bump the version of this file

Comment (oblodgett) [reply src/main/resources/db/migration/v0.38.0.3__gaf-load.sql]: This needs to be turned into a FMS load which will always pull the latest version of the file.

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/v0.38.0.3__gaf-load.sql (hunk 2 above)
Comment (cmpich) [reply src/main/resources/db/migration/v0.38.0.3__gaf-load.sql]: Do we know from where this file gets downloaded from for the loader? Only the human file is referenced from the FMS in the YAML file while all other MODs files have a source URL given at GOC:
https://github.com/alliance-genome/agr_ferret/blob/master/src/datasets/GAF.yaml

Comment (cmpich) [reply src/main/resources/db/migration/v0.38.0.3__gaf-load.sql]: Ok, the human file comes from RGD and that MOD is submitting the generated file into the FMS.

**Review by abecerra — APPROVED** [review]

File: src/main/resources/db/migration/v0.38.0.11__gaf-load.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+update bulkurlload set bulkloadurl = 'https://download.alliancegenome.org/2.3.0/GAF/HUMAN/GAF_HUMAN_1.gaf'
```
Comment (markquintontulloch) [inline src/main/resources/db/migration/v0.38.0.11__gaf-load.sql]: Doesn't this migration need to be deleted now?

Comment (cmpich) [reply src/main/resources/db/migration/v0.38.0.11__gaf-load.sql]: Yes, that is being delete in the next script.

Comment (cmpich) [reply src/main/resources/db/migration/v0.38.0.11__gaf-load.sql]: Actually, I did remove this as it was run second.

Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.38.0.11__gaf-load.sql]: I mean that this whole file can be deleted - it's not doing anything as it's updating an entry that you've already removed

Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.38.0.11__gaf-load.sql]: Sorry, I see you've already removed it!

**Review by markquintontulloch — APPROVED** [review]

**Review by oblodgett — APPROVED** [review]

**Pull request merged**